extensions/amp-lightbox-viewer/0.1/service/lightbox-manager-impl.js

```diff
@@ -120,20 +135,57 @@ export class LightboxManager {
    * @private
    */
   processLightboxElement_(element) {
-    if (element.tagName == 'AMP-CAROUSEL') {
-      // TODO: special case carousel
+    if (element.tagName.toLowerCase() == CAROUSEL_TAG) {
+      let lightboxGroupId = element.getAttribute('lightbox');
+      if (!lightboxGroupId) {
+        lightboxGroupId = 'carousel'
+          + (element.getAttribute('id') || this.counter_++);
+      }
+      this.getSlidesFromCarousel_(element).forEach(slide => {
+        // TODO: review naming conventions for component attributes
+        if (!slide.hasAttribute('lightbox-exclude')) {
+          slide.setAttribute('lightbox', lightboxGroupId);
+          this.processBaseLightboxElement_(slide, lightboxGroupId);
+        }
+      });
     } else {
       const lightboxGroupId = element.getAttribute('lightbox') || 'default';
-      if (!this.lightboxGroups_[lightboxGroupId]) {
-        this.lightboxGroups_[lightboxGroupId] = [];
-      }
-      this.lightboxGroups_[lightboxGroupId]
-          .push(dev().assertElement(element));
-      if (this.meetsHeuristicsForTap_(element)) {
-        const viewer = elementByTag(this.ampdoc_.getRootNode(), VIEWER_TAG).id;
-        element.setAttribute('on', 'tap:' + viewer.id + '.activate');
-      }
+      this.processBaseLightboxElement_(element, lightboxGroupId);
+    }
+  }
+
+  processBaseLightboxElement_(element, lightboxGroupId) {
+    if (!this.lightboxGroups_[lightboxGroupId]) {
+      this.lightboxGroups_[lightboxGroupId] = [];
+    }
+    this.lightboxGroups_[lightboxGroupId]
+        .push(dev().assertElement(element));
+    if (this.meetsHeuristicsForTap_(element)) {
+      const viewer = elementByTag(this.ampdoc_.getRootNode(), VIEWER_TAG);
+      element.setAttribute('on', 'tap:' + viewer.id + '.activate');
+    }
+  }
+
+  /**
+   * @param {!Element} element
+   * @return {!Array<!Element>}
+   * @private
+   */
+  getSlidesFromCarousel_(element) {
+    const type = element.getAttribute('type');
```

Comment:
I think for both types of carousels, each slide has `amp-carousel-slide` class attached to it. So this could become
```
toArray(element.getElementsByClassname('amp-carousel-slide');
```

I think there is also a race-condition here. There is no gaurantee that carousel has finish building by the time it is discovered. so we would need to wait for the "LOAD_END" signal of carousel before querying it. Something like element.whenSignal(CommonSignals.LOAD_END).then should do.